### DISCUSSION TITLE
addr_full versus street_address

### DISCUSSION THREAD

**Discussion by davidhicks** [alltheplaces/alltheplaces#8308] [Q&A] [answered]
Hi @Cj-Malone in #4878 it is suggested that `addr_full` fields be replaced with `street_address`.

[addr:full](https://wiki.openstreetmap.org/wiki/Key:addr:*) is the only OSM key I can find that allows free text entry of an address of any specificity from "The house on the left after you go past the tree on Wood Lane in Somethingshire" to "Suite 2, Level 15, 400 Something St, Townytown, Beststate, Somecountry, 50000" (50000 being a postcode).

Currently ATP spiders appear to mostly be using `addr_full` for any free text entry that includes mention of anything broader than a street (such as mentioning a town, state, region, postcode), and `street_address` when the free text entry is specific just to a particular street. Is there any benefit of keeping this distinction, or is the intent of #4878 to roll both use cases together, into a single ATP key that corresponds to OSM's `addr:full` key?

**Comment by Cj-Malone** [comment, marked as answer]
#4878 is referring to older spiders that output `addr_full` for a street address. This is because `street_address` is significantly newer than `addr_full`. I still have a list of 192 spiders to review.

When spiders have a full address, they should be using `addr_full`, it is not deprecated.

In OSM `addr:full` is a bit of a temporary tag, at least in countries with formal addresses. They should be parsed into the more usable address tags (`addr:unit`, `addr:housenumber`, `addr:housename`, `addr:street`, `addr:city`, `addr:postcode` etc) where possible.

**Reply by iandees** [reply]
... and `street_address` is an All the Places-ism because it can be difficult to reliably parse out a street address like "123 Main St" into `addr:housenumber=123` and `addr:street=Main St`.
